SmartDeviceLink/public/SDLLifecycleConfiguration.m

```diff
@@ -125,6 +117,10 @@ - (void)setAppType:(nullable SDLAppHMIType)appType {
  *  @return            An `appID` made of the first 10 non-dash characters of the "fullAppID"
  */
 + (NSString *)sdl_shortAppIdFromFullAppId:(NSString *)fullAppId {
+    // Do not generate a short app id unless the fullAppID is greater than 10 characters
+    if (fullAppId.length <= AppIdCharacterCount) {
+        return @"";
```

Comment:
Shouldn't this just return the app id? Why an empty string? Wouldn't this break head units that only look at the `appId` when using a Ford-style app id?

Reply:
The `init`s only allow you to set a `fullAppId` and `appId` is `readonly` so there is no way to set an `appId` currently. 

I did have it set to just set the `fullAppId` as the `appId` but you seemed to think it would break things.

Reply:
I think there's a misunderstanding going on. Right now this is very broken. If you set a short app ID (like our Antelope test one) then connect to a head unit that doesn't support `fullAppID` (like a Sync Gen 3) this will not connect because this turns the app id into a blank string.

Reply:
I changed this so:
1. If the `fullAppId` is a valid UUID, it is used to generate the shorter `appId`.
1. If the `fullAppId` is not a valid UUID, then `appId` is set to the same value as the `fullAppId`.